modules/rtdModule/index.js

```diff
@@ -97,61 +172,74 @@ export function init(config) {
   });
 }
 
+/**
+ * call each sub module init function by config order
+ * if no init function / init return failure / module not configured - remove it from submodules list
+ */
+export function initSubModules(subModules) {
+  let subModulesByOrder = [];
+  _dataProviders.forEach(provider => {
+    const sm = subModules.find(s => s.name === provider.name);
+    const initResponse = sm && sm.init && sm.init(provider, gdprDataHandler.getConsentData(), uspDataHandler.getConsentData()) !== 'failure';
```

Comment:
To clarify on the GDPR and USP data, are we sure that in the sequence of things that this data would be available by the time the submodule's `init` functions are called?  

This `initSubModules` function is loaded as part of the overall rtdModule's `init()`, which in turn is loaded as the file is loaded.

But the consent information is obtained once the `requestBids` call chain starts, which should be after this code here has already executed.  So could there be a scenario here where these consent params in the `sm.init` wouldn't have the real data?  Or am I misreading something?

Reply:
Yes, you are correct
I added a hook so it will run before 'makeBidRequests', if you think of a better solution please let me know